taca_ngi_pipeline/deliver/deliver_dds.py

```diff
@@ -0,0 +1,546 @@
+"""
+    Module for controlling deliveries os samples and projects to DDS
+"""
+import requests
+import os
+import logging
+import json
+import subprocess
+import sys
+import re
+
+from ngi_pipeline.database.classes import CharonSession
+from taca.utils.filesystem import create_folder
+from taca.utils.config import CONFIG
+from taca.utils.statusdb import StatusdbSession, ProjectSummaryConnection
+
+from .deliver import ProjectDeliverer, SampleDeliverer, DelivererInterruptedError
+from ..utils.database import DatabaseError
+
+logger = logging.getLogger(__name__)
+
+
+def proceed_or_not(question):
+    yes = set(['yes', 'y', 'ye'])
+    no = set(['no', 'n'])
+    sys.stdout.write("{}".format(question))
+    while True:
+        choice = input().lower()
+        if choice in yes:
+            return True
+        elif choice in no:
+            return False
+        else:
+            sys.stdout.write("Please respond with 'yes' or 'no' ")
+
+
+class DDSProjectDeliverer(ProjectDeliverer):
+    """ This object takes care of delivering project samples with DDS.
+    """
+    def __init__(self, projectid=None, sampleid=None, 
+                 pi_email=None, pi_name=None, sensitive=True,
+                 add_user=None, fcid=None, do_release=False, 
+                 project_title=None, project_description=None,
+                 **kwargs):
+        super(DDSProjectDeliverer, self).__init__(
+            projectid,
+            sampleid,
+            **kwargs
+        )
+        self.config_statusdb = CONFIG.get('statusdb', None)
+        if self.config_statusdb is None and not do_release:
+            raise AttributeError("statusdb configuration is needed when delivering to DDS (url, username, password, port")
+        self.orderportal = CONFIG.get('order_portal', None)
+        if self.orderportal is None and not do_release:
+            raise AttributeError("Order portal configuration is needed when delivering to DDS")
+        if self.orderportal:
+            self._set_pi_details(pi_email, pi_name)
+            self._set_other_member_details(add_user, CONFIG.get('add_project_owner', False))
+            self._set_project_details(project_title, project_description)
+        self.sensitive = sensitive
+        self.fcid = fcid
+        
+    def get_delivery_status(self, dbentry=None):
+        """ Returns the delivery status for this project. If a dbentry
+        dict is supplied, it will be used instead of fethcing from database
+
+        :params dbentry: a database entry to use instead of
+        fetching from db
+        :returns: the delivery status of this project as a string
+        """
+        dbentry = dbentry or self.db_entry()
+        if dbentry.get('delivery_token'):
```

Comment:
Not sure if DDS returns a delivery token, have to check and figure out what to do with this fiel.d

Reply:
Nope, it doesn't. For now I've just set the token to 'uploaded' when all the data has been uploaded successfully, as a way to keep track, similar to how it's done for grus. But we could rethink this in the future and make it more meaningful if we want.